src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -1343,5 +1344,23 @@ public class C : A {
             projectForBaseType = solution2.GetProject(classC.BaseType.ContainingAssembly);
             Assert.Equal(pid1, projectForBaseType.Id);
         }
+
+        [WorkItem(1088127, "DevDiv")]
+        [Fact]
+        public void TestEncodingRetainedAfterTreeChanged()
+        {
+            var ws = new AdhocWorkspace();
+            var proj = ws.AddProject("proj", LanguageNames.CSharp);
+            var doc = ws.AddDocument(proj.Id, "a.cs", SourceText.From("public class c { }", Encoding.UTF32));
+
+            Assert.Equal(Encoding.UTF32, doc.GetTextAsync().Result.Encoding);
```

Comment:
Its okay to use compare with .Equals, so Assert.Equal is okay but Assert.Same is not.